_examples/real-world-examples/server-sent-events/server/event_handlers.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/message/router/middleware"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+)
+
+const (
+	PostCreatedTopic = "post-created"
+	PostUpdatedTopic = "post-updated"
+	FeedUpdatedTopic = "feed-updated"
+)
+
+func SetupMessageRouter(
+	feedsStorage FeedsStorage,
+	logger watermill.LoggerAdapter,
+) (message.Publisher, message.Subscriber, error) {
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	router.AddMiddleware(middleware.Recoverer)
+
+	pubsub := gochannel.NewGoChannel(gochannel.Config{}, logger)
+
+	publishEvents := func(ctx context.Context, tags []string) (messages []*message.Message, err error) {
+		defer func() {
+			if err == nil {
+				logger.Info("Updated posts in feeds", nil)
+			} else {
+				logger.Error("Error in handler", err, nil)
+			}
+		}()
+
+		for _, tag := range tags {
+			logger.Info("Producing event", watermill.LogFields{"tag": tag})
+			event := FeedUpdated{
+				Name:       tag,
+				OccurredAt: time.Now().UTC(),
+			}
+
+			payload, err := json.Marshal(event)
+			if err != nil {
+				return nil, err
+			}
+
+			newMessage := message.NewMessage(watermill.NewUUID(), payload)
+
+			messages = append(messages, newMessage)
+		}
+
+		return messages, nil
+	}
+
+	router.AddHandler(
+		"on-post-created",
+		PostCreatedTopic,
+		pubsub,
+		FeedUpdatedTopic,
+		pubsub,
+		func(msg *message.Message) (messages []*message.Message, err error) {
+			defer func() {
+				if err == nil {
+					logger.Info("Event handler on-post-created executed successfully", nil)
+				} else {
+					logger.Error("Error in on-post-created event handler", err, nil)
+				}
+			}()
+
+			event := PostCreated{}
+			err = json.Unmarshal(msg.Payload, &event)
+			if err != nil {
+				return nil, err
+			}
+
+			logger.Info("Adding post", watermill.LogFields{"post": event.Post})
+
+			if len(event.Post.Tags) > 0 {
+				for _, tag := range event.Post.Tags {
+					logger.Info("Adding tag", watermill.LogFields{"tag": tag})
+					err = feedsStorage.Add(msg.Context(), tag)
```

Comment:
by storage you mean read model? so maybe it's better to use read model name to make it more clear?

Reply:
Feeds don't have a write model, they are essentially the read model for posts. I'm not sure what would be better, `postsReadModel` and `postsWriteModel` seem to not hold the information that the latter is a feed. Also, I'm worried that "model" could be confused with a struct in `models.go`. Any ideas?

Reply:
isn't mysql write model in that case?

Reply:
aah ok I got it now